vector/src/test/java/im/vector/app/features/location/LocationTrackerTest.kt

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.location
+
+import android.content.Context
+import android.location.Location
+import android.location.LocationManager
+import im.vector.app.core.utils.Debouncer
+import im.vector.app.core.utils.createBackgroundHandler
+import im.vector.app.test.fakes.FakeContext
+import im.vector.app.test.fakes.FakeHandler
+import im.vector.app.test.fakes.FakeLocationManager
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.mockkConstructor
+import io.mockk.mockkStatic
+import io.mockk.runs
+import io.mockk.slot
+import io.mockk.unmockkConstructor
+import io.mockk.unmockkStatic
+import io.mockk.verify
+import io.mockk.verifyOrder
+import org.amshove.kluent.shouldBeEqualTo
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+
+private const val A_LATITUDE = 1.2
+private const val A_LONGITUDE = 44.0
+private const val AN_ACCURACY = 5.0f
+
+class LocationTrackerTest {
+
+    private val fakeHandler = FakeHandler()
+
+    init {
+        mockkConstructor(Debouncer::class)
+        every { anyConstructed<Debouncer>().cancelAll() } just runs
+        val runnable = slot<Runnable>()
+        every { anyConstructed<Debouncer>().debounce(any(), MIN_TIME_TO_UPDATE_LOCATION_MILLIS, capture(runnable)) } answers {
+            runnable.captured.run()
+            true
+        }
+        mockkStatic("im.vector.app.core.utils.HandlerKt")
+        every { createBackgroundHandler(any()) } returns fakeHandler.instance
+    }
+
+    private val fakeLocationManager = FakeLocationManager()
+    private val fakeContext = FakeContext().also {
+        it.givenService(Context.LOCATION_SERVICE, android.location.LocationManager::class.java, fakeLocationManager.instance)
+    }
+
+    private val locationTracker = LocationTracker(fakeContext.instance)
+
+    @Before
+    fun setUp() {
+        fakeLocationManager.givenRemoveUpdates(locationTracker)
+    }
+
+    @After
+    fun tearDown() {
```

Comment:
Maybe you should unmockk the objects after each tests ? 
Location or LocationTracker for exemple ?

Reply:
> Maybe you should unmockk the objects after each tests ? Location or LocationTracker for exemple ?

I am not sure to correctly understand your suggestion. Do you suggest to create a new `LocationTracker` object for each test? I didn't think about it but if so, I agree so that we can keep each test isolated from each other.

Reply:
I have done an update here: [abe7fe9](https://github.com/vector-im/element-android/pull/5923/commits/abe7fe917615b588a1e932ce6a30373cda9312c3)